internal/ast/diagnostic.go

```diff
@@ -60,6 +60,18 @@ func (d *Diagnostic) AddRelatedInfo(relatedInformation *Diagnostic) *Diagnostic
 	return d
 }
 
+func (d *Diagnostic) Clone() *Diagnostic {
+	return &Diagnostic{
```

Comment:
Seems like this could just be:

```go
cpy := *d
return &cpy
```

Since no fields have been omitted.

Reply:
Agreed.